src/macad_gym/core/maps/nav_utils.py

```diff
@@ -290,56 +296,48 @@ def generate_path(self):
                 get_next_waypoint(self.world, self.destination)))
         self.path_index = 0
 
-    def advance_path(self):
+    def advance_path_index(self):
         if self.path_index < len(self.path):
-            last_dist = self.path[self.path_index][0].transform.location.\
-                distance(self.actor.get_location())
-
             for i in range(self.path_index + 1, len(self.path)):
-                dist = self.path[i][0].transform.location.\
-                    distance(self.actor.get_location())
-                if dist <= last_dist:
+                index_dist = self.actor.get_location().\
+                    distance(self.path[self.path_index][0].transform.location)
+                next_index_dist = self.actor.get_location().\
+                    distance(self.path[i][0].transform.location)
+                step_dist = self.path[self.path_index][0].transform.location.\
+                    distance(self.path[i][0].transform.location)
+                if step_dist <= index_dist and index_dist > next_index_dist:
                     self.path_index = i
                 else:
+                    if step_dist >= next_index_dist:
+                        self.path_index = i+1 if i+1 < len(self.path) else i
                     break
-                last_dist = dist
-
-    def seek_closest(self):
-        if len(self.path) > 0:
-            close_dist = self.path[0][0].transform.location.\
-                distance(self.actor.get_location())
-            close_i = 0
 
-            for i in range(1, len(self.path)):
-                cur_dist = self.path[i][0].transform.location.\
-                    distance(self.actor.get_location())
-                if cur_dist <= close_dist:
-                    close_i = i
-
-            self.path_index = close_i
+    def seek_next_waypoint(self):
+        assert len(self.path) > 0, "No waypoints in path list."
+        i = 0
+        for i in range(1, len(self.path)):
+            index_dist = self.actor.get_location().\
+                distance(self.path [ i-1][0].transform.location)
+            step_dist = self.path [ i-1][0].transform.location. \
+                distance(self.path[i][0].transform.location)
+            if step_dist > index_dist and i+1 < len(self.path):
+                i += 1
+                break
+        self.path_index = i
```

Comment:
Could you explain what this change addresses or improves and why?
Adding docstrings will help too.